src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,483 @@
+                    throw new RpcException(-100, "Request is not found");
+                var data = oraclePub.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+                if (!Crypto.VerifySignature(data, msgSign, oraclePub)) throw new RpcException(-100, "Invalid sign");
+
+                AddResponseTxSign(snapshot, requestId, oraclePub, txSign);
+            }
+            return new JObject();
+        }
+
+        private static async Task SendContentAsync(string url, string content)
+        {
+            try
+            {
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+                request.Method = "POST";
+                request.ContentType = "application/json";
+                request.Timeout = 5000;
+                using (StreamWriter dataStream = new StreamWriter(await request.GetRequestStreamAsync()))
+                {
+                    await dataStream.WriteAsync(content);
+                }
+                HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
+                if (response.ContentLength > ushort.MaxValue) throw new Exception("The response it's bigger than allowed");
+                StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.UTF8);
+                await reader.ReadToEndAsync();
+            }
+            catch (Exception e)
+            {
+                Log($"Failed to send the response signature to {url}, as {e.Message}", LogLevel.Warning);
+            }
+        }
+
+        private async Task SendResponseSignatureAsync(ulong requestId, byte[] txSign, KeyPair keyPair)
+        {
+            var message = keyPair.PublicKey.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            var sign = Crypto.Sign(message, keyPair.PrivateKey, keyPair.PublicKey.EncodePoint(false)[1..]);
+            var param = "\"" + Convert.ToBase64String(keyPair.PublicKey.ToArray()) + "\", " + requestId + ", \"" + Convert.ToBase64String(txSign) + "\",\"" + Convert.ToBase64String(sign) + "\"";
+            var content = "{\"id\":" + Interlocked.Increment(ref counter) + ",\"jsonrpc\":\"2.0\",\"method\":\"submitoracleresponse\",\"params\":[" + param + "]}";
+
+            var tasks = Settings.Default.Nodes.Select(p => SendContentAsync(p, content));
+            await Task.WhenAll(tasks);
+        }
+
+        private async Task ProcessRequestAsync(StoreView snapshot, OracleRequest req)
+        {
+            Log($"Process oracle request: {req}, txid: {req.OriginalTxid}, url: {req.Url}");
+
+            (OracleResponseCode code, string data) = await ProcessUrlAsync(req.Url);
+
+            foreach (var (requestId, request) in NativeContract.Oracle.GetRequestsByUrl(snapshot, req.Url))
+            {
+                var result = code == OracleResponseCode.Success ? Filter(data, request.Filter) : Array.Empty<byte>();
```

Comment:
The exception thrown by `Filter()` should be caught.

Reply:
Done